daemons/fenced/fenced_commands.c

```diff
@@ -3351,6 +3353,36 @@ handle_unknown_request(pcmk__request_t *request)
     return fenced_construct_reply(request->xml, NULL, &request->result);
 }
 
+static void
+fenced_register_handlers(void)
+{
+    pcmk__server_command_t handlers[] = {
+        { CRM_OP_REGISTER, handle_register_request },
+        { STONITH_OP_EXEC, handle_agent_request },
+        { STONITH_OP_TIMEOUT_UPDATE, handle_update_timeout_request },
+        { STONITH_OP_QUERY, handle_query_request },
+        { T_STONITH_NOTIFY, handle_notify_request },
+        { STONITH_OP_RELAY, handle_relay_request },
+        { STONITH_OP_FENCE, handle_fence_request },
+        { STONITH_OP_FENCE_HISTORY, handle_history_request },
+        { STONITH_OP_DEVICE_ADD, handle_device_add_request },
+        { STONITH_OP_DEVICE_DEL, handle_device_delete_request },
+        { STONITH_OP_LEVEL_ADD, handle_level_add_request },
+        { STONITH_OP_LEVEL_DEL, handle_level_delete_request },
+        { CRM_OP_RM_NODE_CACHE, handle_cache_request },
+        { NULL, handle_unknown_request },
+    };
```

Comment:
pcmk__server_command_t is a bit superfluous; we could just use a hash table directly, like

    g_hash_table_insert(fenced_handlers, CRM_OP_REGISTER, handle_register_request);
    g_hash_table_insert(fenced_handlers, STONITH_OP_EXEC, handle_agent_request);
    g_hash_table_insert(fenced_handlers, STONITH_OP_TIMEOUT_UPDATE, handle_update_timeout_request);
    // etc.

but I felt the array looked less cluttered. I'm not committed to it though.

Reply:
and it opens possibility to play with the implementation without having to change the sub-daemon-code.
tried to play with some ideas in other comments.